src/core/moves/proposal/tree/SubtreePruneRegraftProposal.cpp

```diff
@@ -94,12 +94,23 @@ bool SubtreePruneRegraftProposal::isDescendant(const TopologyNode &n, const Topo
  */
 double SubtreePruneRegraftProposal::doProposal( void )
 {
+    // reset flag
+    failed = false;
     
     // Get random number generator
     RandomNumberGenerator* rng     = GLOBAL_RNG;
     
     Tree& tau = tree->getValue();
     
+    // when the tree has fewer than 4 tips, nothing should be done
+    size_t num_tips = tau.getNumberOfTips();
+    size_t num_root_children = tau.getRoot().getNumberOfChildren();
+    if ( num_tips <= num_root_children)
```

Comment:
Why do you need to test the root children and not only test for num_tips <= 4?

Reply:
Please see the response above to the previous comment on the same code.